layers/b2qt/recipes-qt/automotive/neptune-ui_git.bbappend

```diff
@@ -10,3 +10,40 @@ RDEPENDS_${PN}_append = "\
       qtwebengine        \
       qtquickcontrols2   \
       "
+
+HAS_CONTAINMENT = "${@bb.utils.contains('DISTRO_FEATURES', 'process-containment', 'TRUE', 'FALSE', d)}"
+
+#
+# Add software-container AM config to appman cmdline if we have containment
+# support
+#
+do_install_prepend() {
+    echo "Containment: ${HAS_CONTAINMENT}"
+    SC_CONFIG_LINE="-c /opt/am/sc-config.yaml"
+    SERVICEFILE="${WORKDIR}/neptune.service"
+
+
+    # We only add it if we have containment and it isn't already there
+    if [ "${HAS_CONTAINMENT}" = "TRUE" ]; then
+        if  ! grep -q "sc-config.yaml" $SERVICEFILE; then
```

Comment:
If someone knows a better way to grep for SC_CONFIG_LINE (it interprets the -c etc in a bad way), I would prefer that.

Reply:
How about adding $__EXTRA_ARGUMENTS__ at the end of ExecStart in the service file, which would be resolved as an empty variable by default.

In your SED line you could then replace $__EXTRA_ARGUMENTS__ with your config + $__EXTRA_ARGUMENTS__, which would also enable other recipes to add extra options ?

In addition i'm not sure whether it's worth to check whether sc-config.yaml is already part of the line, if it added twice, it doesn't matter, as appman shouldn't have a problem with that and if a user has it's custom container-config.yaml instead it won't work anyway ?

Reply:
I like the idea of using $EXTRA_ARGUMENTS actually, as it can make this much easier, one can set its value already in the HAS_CONTAINER variable and then just replace it in the sed script. And if it is already replaced then the sed line won't find $EXTRA_ARGUMENTS anyway and we're good.